syft/frameworks/crypten/message_handler.py

```diff
@@ -66,24 +65,39 @@ def run_crypten_party_jail(self, msg: CryptenInitJail):
             An ObjectMessage containing the return value of the crypten function computed.
         """
 
-        self.worker.rank_to_worker_id, world_size, master_addr, master_port = msg.crypten_context
+        rank_to_worker_id, world_size, master_addr, master_port = msg.crypten_context
+
+        cid = CryptenMessageHandler._get_computation_id()
+        syft_crypten.RANK_TO_WORKER_ID[cid] = rank_to_worker_id
 
         ser_func = msg.jail_runner
         onnx_model = msg.model
         crypten_model = None if onnx_model is None else utils.onnx_to_crypten(onnx_model)
         jail_runner = JailRunner.detail(ser_func, model=crypten_model)
 
-        rank = self._current_rank()
+        rank = self._current_rank(rank_to_worker_id)
         assert rank is not None
 
-        return_value = run_party(jail_runner, rank, world_size, master_addr, master_port, (), {})
+        return_value = run_party(
+            cid, jail_runner, rank, world_size, master_addr, master_port, (), {}
+        )
+        # remove rank to id transaltion dict
+        del syft_crypten.RANK_TO_WORKER_ID[cid]
+
         return ObjectMessage(return_value)
 
-    def _current_rank(self):
+    def _current_rank(self, rank_to_worker_id):
         """Returns current rank based on worker_id."""
         rank = None
-        for r, worker_id in self.worker.rank_to_worker_id.items():
+        for r, worker_id in rank_to_worker_id.items():
             if worker_id == self.worker.id:
                 rank = r
                 break
         return rank
+
+    @staticmethod
+    def _get_computation_id():
+        cid = randint(0, 2 ** 32)
```

Comment:
Q: Can't we use the ```id provider``` from syft?

Reply:
if it's okay to consume IDs from their than I'm okay with it, however, we can have overlapping IDs in the current implementation without any problem, none of them depends on the other.

Reply:
Yes I guess we should use the id provider!

Reply:
And you can brun man of them if need as you do with the wile loop